blueprints/networking/psc-glb-and-armor/consumer.tf

```diff
@@ -14,125 +14,124 @@
  * limitations under the License.
  */
 
-module "consumer_project" {
+module "consumer-project" {
   source         = "../../../modules/project"
   name           = var.consumer_project_id
-  project_create = var.project_create
+  project_create = var.project_create_config != null
+
+  billing_account = try(var.project_create_config.billing_account)
+  parent          = try(var.project_create_config.parent)
+  prefix          = var.prefix
   services = [
     "iam.googleapis.com",
     "compute.googleapis.com",
   ]
 }
 
-module "producer_a_project" {
-  source              = "./modules/producer"
-  producer_project_id = var.producer_a_project_id
-  project_create      = var.project_create
-}
-
-module "producer_b_project" {
-  source              = "./modules/producer"
-  producer_project_id = var.producer_b_project_id
-  project_create      = var.project_create
-}
-
-resource "google_compute_region_network_endpoint_group" "psc_neg_a" {
-  name                  = "psc-neg-a"
-  region                = var.region
-  project               = module.consumer_project.project_id
-  network_endpoint_type = "PRIVATE_SERVICE_CONNECT"
-  psc_target_service    = module.producer_a_project.psc_ilb_service_attachment.self_link
-
-  network    = "default"
-  subnetwork = "default"
-}
-
-resource "google_compute_region_network_endpoint_group" "psc_neg_b" {
-  name                  = "psc-neg-b"
-  region                = var.region
-  project               = module.consumer_project.project_id
-  network_endpoint_type = "PRIVATE_SERVICE_CONNECT"
-  psc_target_service    = module.producer_b_project.psc_ilb_service_attachment.self_link
-
-  network    = "default"
-  subnetwork = "default"
+module "producer-a" {
+  source                = "./modules/producer"
+  consumer_project_id   = module.consumer-project.project_id
+  prefix                = var.prefix
+  producer_project_id   = var.producer_a_project_id
+  project_create_config = var.project_create_config
```

Comment:
Missing setting of `region`, so producer resources are crated in different region and it breaks the blueprint.